src/pybind/mgr/cephadm/schedule.py

```diff
@@ -216,18 +216,20 @@ def remove_daemon_hosts(self, host_pool: List[HostPlacementSpec]) -> Set[DaemonD
         return _remove_daemon_hosts
 
     def get_candidates(self) -> List[HostPlacementSpec]:
+        per_host = self.spec.placement.get_num_per_host()
+
         if self.spec.placement.hosts:
-            return self.spec.placement.hosts
-        elif self.spec.placement.label:
+            return self.spec.placement.hosts * per_host
```

Comment:
That doesn't work. this has to be more complicated: 

```yaml
placement:
  hosts:    
  - mon-host-a=a
  - mon-host-b=b
  - mon-host-c=c
  max_per_host=3
```
would create garbage. 

For explicit host placement, we already have a way to allow explicit co-location:

```yaml
placement:
  hosts:    
  - mon-host-a=a
  - mon-host-a=b
  - mon-host-a=c
```

I'd be :+1: for making `max_per_host` incompatible with explicit host placement for now